src/pybamm/solvers/processed_variable.py

```diff
@@ -458,6 +462,26 @@ def initialise_sensitivity_explicit_forward(self):
 
             # Compute sensitivity
             S_var = dvar_dy_eval @ dy_dp + dvar_dp_eval
+
+            # post fix for discrete time integral won't give correct result
+            # if ts are not equal to the discrete times. Raise error
+            # in this case
+            if (
+                self.time_integral is not None
+                and self.time_integral.method == "discrete"
+            ):
+                if not (
+                    len(ts) == len(self.time_integral.discrete_times)
+                    and np.all(np.abs(ts - self.time_integral.discrete_times) < 1e-10)
```

Comment:
Seems like a good spot to use allclose 

```suggestion
                    and np.allclose(ts, self.time_integral.discrete_times, atol=1e-10))
```